core/autoscaling_test.go

```diff
@@ -3353,6 +3412,79 @@ func TestGetDisallowedInstanceTypes(t *testing.T) {
 	}
 }
 
+func TestFilteringCompleteSIRRequests(t *testing.T) {
+	tests := []struct {
+		foundSIRRequests    []*ec2.SpotInstanceRequest
+		expectedSIRRequests []*ec2.SpotInstanceRequest
+	}{
+		{
+			foundSIRRequests: []*ec2.SpotInstanceRequest{
+				{
+					SpotInstanceRequestId: aws.String("bidTestId1"),
+					Tags: []*ec2.Tag{
+						{Key: aws.String("launched-for-asg"), Value: aws.String("xyz")},
+						{Key: aws.String("autospotting-complete"), Value: aws.String("true")},
+					},
+				},
+				{
+					SpotInstanceRequestId: aws.String("bidTestId2"),
+					Tags: []*ec2.Tag{
+						{Key: aws.String("launched-for-asg"), Value: aws.String("xyz")},
+					},
+				},
+			},
+			expectedSIRRequests: []*ec2.SpotInstanceRequest{
+				{
+					SpotInstanceRequestId: aws.String("bidTestId2"),
+					Tags: []*ec2.Tag{
+						{Key: aws.String("launched-for-asg"), Value: aws.String("xyz")},
+					},
+				},
+			},
+		},
+		{
+			foundSIRRequests: []*ec2.SpotInstanceRequest{
+				{
+					SpotInstanceRequestId: aws.String("bidTestId1"),
+					Tags: []*ec2.Tag{
+						{Key: aws.String("launched-for-asg"), Value: aws.String("xyz")},
+						{Key: aws.String("autospotting-complete"), Value: aws.String("true")},
+					},
+				},
+				{
+					SpotInstanceRequestId: aws.String("bidTestId2"),
+					Tags: []*ec2.Tag{
+						{Key: aws.String("launched-for-asg"), Value: aws.String("xyz")},
+						{Key: aws.String("autospotting-complete"), Value: aws.String("true")},
+					},
+				},
+			},
+			expectedSIRRequests: []*ec2.SpotInstanceRequest{},
+		},
+		{
+			foundSIRRequests:    []*ec2.SpotInstanceRequest{},
+			expectedSIRRequests: []*ec2.SpotInstanceRequest{},
+		},
+		{
+			foundSIRRequests:    nil,
+			expectedSIRRequests: []*ec2.SpotInstanceRequest{},
+		},
+	}
+
+	for _, tt := range tests {
+		filteredSIRRequests := filterOutCompleteSpotInstanceRequests(tt.foundSIRRequests)
+		if len(filteredSIRRequests) == 0 {
```

Comment:
Would that trigger an error if it wasn't 0?

Reply:
Have commented in test